server/resources.py

```diff
@@ -12,39 +43,61 @@ def post(self):
 
         new_user = UserModel(
             email = data['email'],
-            password = UserModel.generate_hash(data['password'])
+            password = data['password']
         )
         try:
             new_user.save_to_db()
-            return {'message': 'User {} was created'.format(data['email'])}, 201
+            auth_token = UserModel.encode_auth_token(new_user.id)
+            responseObject = {
+                'status': 'success',
+                'message': 'User {} was created'.format(data['email']),
+                'auth_token': auth_token.decode()
+            }
+            return responseObject, 201
         except Exception as e:
-            return {'message': 'Something went wrong'}, 500
+            responseObject = {
+                'status': 'fail',
+                'message': 'Something went wrong. Please try again.'
+            }
+            return responseObject, 500
 
 
 
 class UserLogin(Resource):
     def post(self):
         data = user_schema.load(request.json)
         current_user = UserModel.find_by_email(data['email'])
-        if not current_user:
-            return {'message': 'User {} doesn\'t exist'.format(data['email'])}
+        pw_is_valid = flask_bcrypt.check_password_hash(current_user.password, data.get('password'))
```

Comment:
can we move the bcrypt logic into the User model (I think it was there before, there was verify password method on the user)

Reply:
i'll look into that!